### PR TITLE
Always Collect Device Data

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

 -  Remove `collectDeviceData` from `DropInRequest` in favor of always returning device data

 ### Checklist

 - [x] Added a changelog entry

- @sarahkoop

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Braintree Android Drop-In Release Notes
 
+## unreleased
+* Breaking Changes
+  * DropInRequest
+    * Remove `collectDeviceData` from `DropInRequest` in favor of always returning device data
```
Comment (scannillo) [inline CHANGELOG.md:6]: Maybe we update this to mention where the device data gets returned (BTDropInResult)?

Comment (sarahkoop) [reply CHANGELOG.md]: Yeah that sounds more clear - updated.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Braintree Android Drop-In Release Notes
 
+## unreleased
+* Breaking Changes
+  * DropInRequest
+    * Remove `collectDeviceData` from `DropInRequest` in favor of always returning device data in `DropInResult`
```
Comment (scannillo) [inline CHANGELOG.md:6]:
```suggestion
    * Remove `collectDeviceData` from `DropInRequest` in favor of always returning device data on `DropInResult`
```

File: Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java [hunk 3]
```diff
@@ -528,23 +528,18 @@ public void onResult(@Nullable DropInResult dropInResult, @Nullable Exception er
...
-                                if (deviceData != null) {
-                                    dropInResult.deviceData(deviceData);
-                                    animateBottomSheetClosedAndFinishDropInWithResult(dropInResult);
-                                } else {
-                                    updateVaultedPaymentMethodNonces(true);
-                                    onError(error);
-                                }
+                    getDropInClient().collectDeviceData(DropInActivity.this, new DataCollectorCallback() {
+                                                                          @Override
+                                                                          public void onResult(@Nullable String deviceData, @Nullable Exception error) {
```
Comment (scannillo) [inline Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java:533]: Nitpick - but the indentation looks off here for the `override` and `public void` lines

File: Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java [hunk 4]
```diff
@@ -105,22 +105,17 @@ public void onResult(@Nullable ThreeDSecureResult lookupResult, @Nullable Except
...
-                                    public void onResult(@Nullable String deviceData, @Nullable Exception error) {
-                                        if (deviceData != null) {
-                                            dropInResult.deviceData(deviceData);
-                                            callback.onResult(dropInResult, null);
-                                        } else {
-                                            callback.onResult(null, error);
-                                        }
+                            dataCollector.collectDeviceData(activity, new DataCollectorCallback() {
+                                                                  @Override
+                                                                  public void onResult(@Nullable String deviceData, @Nullable Exception error) {
```
Comment (scannillo) [inline Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java:110]: Same indentation comment?

**Review by scannillo — APPROVED** [review]
2 minor comments, looks good!!!!

File: Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java (hunk 3 above)
Comment (sarahkoop) [reply Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java]: Ah yeah those get collapsed in the IDE so didn't notice the weird indent. Should be fixed now.

**Review by sshropshire — APPROVED** [review]
🌊

**Pull request merged**